sw/airborne/subsystems/ins/ins_vectornav.h

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (C) 2015 Michal Podhradsky, michal.podhradsky@aggiemail.usu.edu
+ * Utah State University, http://aggieair.usu.edu/
+ *
+ * This file is part of paparazzi.
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+/**
+ * @file ins_vectornav.h
+ *
+ * Vectornav VN-200 INS subsystem
+ *
+ * @author Michal Podhradsky <michal.podhradsky@aggiemail.usu.edu>
+ */
+#ifndef INS_VECTORNAV_H
+#define INS_VECTORNAV_H
+
+// Subsystems
+#include "subsystems/imu.h"
+#include "subsystems/gps.h"
+#include "subsystems/ahrs.h"
+#include "subsystems/ins.h"
+
+// Peripheral
+#include "peripherals/vn200_serial.h"
+
+// Geodetic / Math
+#include "math/pprz_algebra.h"
+#include "math/pprz_geodetic_int.h"
+#include "math/pprz_algebra_float.h"
+#include "math/pprz_geodetic_int.h"
+#include "math/pprz_isa.h"
+
+// Generated
+#include "generated/airframe.h"
+#include "generated/flight_plan.h"
+
+// Systime
+#include "mcu_periph/sys_time.h"
+
+// Abi
+#include "subsystems/abi.h"
+
+#if GPS_USE_LATLONG
+/* currently needed to get nav_utm_zone0 */
+#include "subsystems/navigation/common_nav.h"
+#include "math/pprz_geodetic_float.h"
+#endif
+
+/*
+ * IMU Subsystem defines
+ */
+extern void imu_vectornav_event(void);
+
+#if !USE_NPS
+#define ImuEvent imu_vectornav_event
+#endif /* Only for non-NPS targets */
+
+/*
+ * IMU scalling defaults - still needed, although not used
+ * Filled in with dummy values (ones)
+ */
+#if !defined IMU_GYRO_P_SENS & !defined IMU_GYRO_Q_SENS & !defined IMU_GYRO_R_SENS
```

Comment:
not needed if you don't add imu.c and register the IMU msgs from this implementation.